aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -36,9 +36,8 @@ struct Indexer {
     for (int j = 0; j < num_indexers; j++) {
       int64_t value = *(int64_t*)&indexers[j][idx * indexer_strides[j]];
       int64_t size = original_sizes[j];
-      if (value < -size || value >= size) {
-        TORCH_CHECK_INDEX(false, "index ", value, " is out of bounds for dimension ", j, " with size ", size);
-      }
+      TORCH_CHECK_INDEX(value >= -size && value < size,
```

Comment:
shouldn't it be `value < -size || value >= size`? or is the `&&` intentional?

Reply:
We should `TORCH_CHECK_INDEX(!(value < -size || value >= size) , ...)`

`!(value < -size || value >= size)` == `!(value < -size) && !(value >= size)`  (DeMorgan's Law)
`!(value < -size) && !(value <= size)` == `value >= -size && value < size`